### DISCUSSION TITLE
Plugin development - provide audio feedback following keypress

### DISCUSSION THREAD

**Discussion by chester108** [jspsych/jsPsych#2211] [General]
Hi everyone - just sharing a solution to a problem I recently faced when trying to set up and play a sound file so it could be used as feedback during a task. The solution is based on an earlier answer: https://github.com/jspsych/jsPsych/discussions/1652

load audio stimuli
```
var stim_sound=[
  '../static/audio/reward_sound.mp3',
  '../static/audio/loss_sound.mp3',
]
```

assign audio files in the array to variables
```
var loss_sound = stim_sound[1];
    
var reward_sound = stim_sound[0];
```

write a function to set up and play the audio file
```
    function _PlaySoundFeedback(feedback_sound){
    audio = new Audio();
    audio.src = feedback_sound;
    audio.loop = false; 
    audio.play();
    }  // when calling the function you can assign either 'loss_sound' or 'reward_sound' to the _PlaySoundFeedback() function
```

preload the audio files within jsPsych.init - 
```
jsPsych.init(
use_webaudio: false, // if you are hosting on your local computer, setting this to false will avoid an security errors
preload_audio: [stim_sound],
...
)
```

**Comment by becky-gilbert** [comment]
Hi @chester108, it looks like the code you posted is setting the value of `display.sound_feedback`, which I assume determines what sound should be played. But to figure out why the sound isn't playing, we'd also need to see the code that you're using to set up and play the sound file. Can you post that too please?

**Comment by becky-gilbert** [comment]
Sorry for being slow to respond, and glad you were able to solve this. Thanks for sharing the solution!
